packages/adapter-cloudflare-workers/files/entry.js

```diff
@@ -11,100 +16,112 @@ const app_path = `/${manifest.appPath}`;
-		let is_static_asset = false;
-		const filename = stripped_pathname.slice(base_path.length + 1);
-		if (filename) {
-			is_static_asset =
-				manifest.assets.has(filename) ||
-				manifest.assets.has(filename + '/index.html') ||
-				filename in manifest._.server_assets ||
-				filename + '/index.html' in manifest._.server_assets;
-		}
+	const stripped_pathname = pathname.replace(/\/$/, '');
+
+	// prerendered pages and /static files
+	let is_static_asset = false;
+	const filename = stripped_pathname.slice(base_path.length + 1);
+	if (filename) {
+		is_static_asset =
+			manifest.assets.has(filename) ||
+			manifest.assets.has(filename + '/index.html') ||
+			filename in manifest._.server_assets ||
+			filename + '/index.html' in manifest._.server_assets;
+	}
 
-		let location = pathname.at(-1) === '/' ? stripped_pathname : pathname + '/';
-
-		if (
-			is_static_asset ||
-			prerendered.has(pathname) ||
-			pathname === version_file ||
-			pathname.startsWith(immutable)
-		) {
-			return get_asset_from_kv(req, env, context, (request, options) => {
-				if (prerendered.has(pathname)) {
-					url.pathname = '/' + prerendered.get(pathname).file;
-					return new Request(url.toString(), request);
-				}
-
-				return mapRequestToAsset(request, options);
-			});
-		} else if (location && prerendered.has(location)) {
-			if (search) location += search;
-			return new Response('', {
-				status: 308,
-				headers: {
-					location
-				}
-			});
-		}
+	let location = pathname.at(-1) === '/' ? stripped_pathname : pathname + '/';
+
+	if (
+		is_static_asset ||
+		prerendered.has(pathname) ||
+		pathname === version_file ||
+		pathname.startsWith(immutable)
+	) {
+		return get_asset_from_kv(req, env, context, (request, options) => {
+			if (prerendered.has(pathname)) {
+				url.pathname = '/' + prerendered.get(pathname).file;
+				return new Request(url.toString(), request);
+			}
 
-		// dynamically-generated pages
-		return await server.respond(req, {
-			platform: {
-				env,
-				context,
-				// @ts-expect-error lib.dom is interfering with workers-types
-				caches,
-				// @ts-expect-error req is actually a Cloudflare request not a standard request
-				cf: req.cf
-			},
-			getClientAddress() {
-				return req.headers.get('cf-connecting-ip');
+			return mapRequestToAsset(request, options);
+		});
+	} else if (location && prerendered.has(location)) {
+		if (search) location += search;
+		return new Response('', {
+			status: 308,
+			headers: {
+				location
 			}
 		});
 	}
-};
+
+	// dynamically-generated pages
+	return await server.respond(req, {
+		platform: {
+			env,
+			context,
+			// @ts-expect-error lib.dom is interfering with workers-types
+			caches,
+			// @ts-expect-error req is actually a Cloudflare request not a standard request
+			cf: req.cf
+		},
+		getClientAddress() {
+			return req.headers.get('cf-connecting-ip');
+		}
+	});
+}
+
+export default 'prototype' in handlers && handlers.prototype instanceof WorkerEntrypoint
```

Comment:
Could we perform this check during the build and only import `WorkerEntrypoint` if it is required? We should also document how to make the SvelteKit worker compatible with RPC.